### PR TITLE
Bug1735

### PR DESCRIPTION
**Pull request by mattwar**
Fix for #1735 

@basoundr @Pilchie  please review.

The smart indenter is not very smart when it comes to the first token in the file. The logic is hard to follow in the GetDesiredIndentation function, (I'm not entirely sure what it all is trying to do), but it definitely should not have been attempting to determine the indentation based on the token in this case, because that function assumes that minimum indentation must occur.

### CODE REVIEW

**Comment by basoundr** [comment]
With this change we will have the following behavior
Initial Code:
```C#
    //comment
class C
{
}
```
After code:
```C#
    //comment

    class C
{
}
```
This is because it takes the same indentation as the previous line.

**Comment by basoundr** [comment]
Had it checked with VS2013 and we have the same behavior where the statement moves to the same indentation as the previous comment. @heejaechang @Pilchie Is this the behavior we desire?

**Comment by heejaechang** [comment]
it looks fine for me.

...

smart indenter is full of heuristic so code is quite messy. it might have been cleaner if I just wrote everything case by case rather than trying to merge conditions to some generic rules.

**Comment by Pilchie** [comment]
Looks fine to me.

Might need to pull master and merge into your branch and then push again to get a clean Jenkins run.
